src/api/source/mvt.js

```diff
@@ -68,70 +74,90 @@ export default class MVT extends Base {
         }
         const tile = new VectorTile(new Protobuf(arrayBuffer));
         const mvtLayer = tile.layers[Object.keys(tile.layers)[0]];
-
-        const { points, featureGeometries, properties } = this._decodeMVTLayer(mvtLayer, this._metadata, MVT_EXTENT);
-
-
+        const { geometries, properties } = this._decodeMVTLayer(mvtLayer, this._metadata, MVT_EXTENT);
         const rs = rsys.getRsysFromTile(x, y, z);
-        const dataframeGeometry = this._metadata.geomType == geometryTypes.POINT ? points : featureGeometries;
-        const dataframe = this._generateDataFrame(rs, dataframeGeometry, properties, mvtLayer.length, this._metadata.geomType);
+        const dataframe = this._generateDataFrame(rs, geometries, properties, mvtLayer.length, this._metadata.geomType);
+
         return dataframe;
     }
 
 
     _decodeMVTLayer(mvtLayer, metadata, mvt_extent) {
-        let points;
-        if (metadata.geomType == geometryTypes.POINT) {
-            points = new Float32Array(mvtLayer.length * 2);
+        if (!metadata.geomType) {
+            metadata.geomType = this._autoDiscoverType(mvtLayer);
         }
-        let featureGeometries = [];
-        const decodedProperties = {};
-        const decodingPropertyNames = Object.keys(metadata.properties).
-            filter(propertyName => metadata.properties[propertyName].type != 'geometry').
-            map(propertyName => metadata.properties[propertyName].sourceName || propertyName);
-        decodingPropertyNames.forEach(propertyName => {
-            decodedProperties[propertyName] = new Float32Array(mvtLayer.length + RTT_WIDTH);
-        });
+        switch (metadata.geomType) {
+            case geometryTypes.POINT:
+                return this._decode(mvtLayer, metadata, mvt_extent, new Float32Array(mvtLayer.length * 2));
+            case geometryTypes.LINE:
+                return this._decode(mvtLayer, metadata, mvt_extent, [], decodeLines);
+            case geometryTypes.POLYGON:
+                return this._decode(mvtLayer, metadata, mvt_extent, [], decodePolygons);
+            default:
+                throw new Error('MVT: invalid geometry type');
+        }
+    }
+
+    _autoDiscoverType(mvtLayer) {
+        const type = mvtLayer.feature(0).type;
```

Comment:
The layer could have zero features

Reply:
wops, fixed [here](https://github.com/CartoDB/carto-vl/pull/626/commits/9031c8e3890714d1dea0a17b4b4d08b998842ca3)